phoenix-core-client/src/main/java/org/apache/phoenix/iterate/OrderedResultIterator.java

```diff
@@ -301,21 +415,51 @@ private PeekingResultIterator getResultIterator() throws SQLException {
                 }
                 queueEntries.add(new ResultEntry(sortKeys, result));
                 if (EnvironmentEdgeManager.currentTimeMillis() - startTime >= pageSizeMs) {
-                    dummyTuple = getDummyTuple(result);
-                    return resultIterator;
+                    return getDummyResult();
                 }
             }
             resultIteratorReady = true;
             this.byteSize = queueEntries.getByteSize();
         } catch (IOException e) {
+            LOGGER.error("Error while getting result iterator from OrderedResultIterator.", e);
             ClientUtil.createIOException(e.getMessage(), e);
+            throw new SQLException(e);
         } finally {
             delegate.close();
         }
         
         return resultIterator;
     }
 
+    /**
+     * Retrieve dummy rowkey and return to the client.
+     *
+     * @return the result iterator.
+     */
+    private PeekingResultIterator getDummyResult() {
```

Comment:
This should not return PeekingResultIterator. Should it be private void updateDummyResult()?